script.js

```diff
@@ -0,0 +1,942 @@
+|| (text === variables.specialKeys.PAGE_DOWN && e.code === 'PageDown')
+|| (text === variables.specialKeys.ALTGR && e.code === 'AltRight')
+|| (text === variables.specialKeys.CTRL && e.code === 'ControlLeft')
+|| (text === variables.specialKeys.SHIFT && e.code === 'ShiftLeft')
+|| (text === variables.specialKeys.SCROLL_LOCK && e.key === 'ScrollLock')
+|| (text === variables.specialKeys.PRINT_SCREEN && e.key === 'PrintScreen')
+|| (text === '<' && e.code === 'IntlBackslash')
+);
+
+const isSecondKey = (text, e) => (
+  (text === variables.specialKeys.CTRL && e.code === 'ControlRight')
+|| (text === variables.specialKeys.SHIFT && e.code === 'ShiftRight')
+|| (text === variables.specialKeys.ALT && e.code === 'AltRight')
+|| (text === variables.specialKeys.ENTER && e.code === 'NumpadEnter')
+|| (text === '\\' && e.code === 'IntlBackslash')
+|| (text === '/' && e.code === 'Slash')
+|| (text === '-' && e.code === 'NumpadSubtract')
+|| (text === '0' && e.code === 'Numpad0')
+|| (text === '1' && e.code === 'Numpad1')
+|| (text === '2' && e.code === 'Numpad2')
+|| (text === '3' && e.code === 'Numpad3')
+|| (text === '4' && e.code === 'Numpad4')
+|| (text === '5' && e.code === 'Numpad5')
+|| (text === '6' && e.code === 'Numpad6')
+|| (text === '7' && e.code === 'Numpad7')
+|| (text === '8' && e.code === 'Numpad8')
+|| (text === '9' && e.code === 'Numpad9')
+|| (text === '.' && e.code === 'NumpadDecimal')
+);
+
+const processKeySelection = (e) => {
+  let isKeydown;
+  switch (e.type) {
+    case 'keydown': isKeydown = true; break;
+    case 'keyup': isKeydown = false; break;
+    default: break;
+  }
+
+  const addRemoveKeyActive = (currentKey) => {
+    if (isKeydown) {
+      currentKey.classList.add(classes.KEY_ACTIVE);
+    } else {
+      currentKey.classList.remove(classes.KEY_ACTIVE);
+    }
+  };
+
+  let isRightKey = false;
+  let target;
+  for (let i = 0; i < elements.keys.length; i += 1) {
+    const key = elements.keys[i];
+    const keyText = key.innerText;
+    if ((keyText === variables.specialKeys.META && e.key === 'Meta')) {
+      addRemoveKeyActive(key);
+      if (isKeydown) {
+        key.firstElementChild.classList.remove(
+          isPlatformWindows() ? classes.META_WIN : classes.META_APPLE,
+        );
+        key.firstElementChild.classList.add(
+          isPlatformWindows() ? classes.META_WIN_ACTIVE : classes.META_APPLE_ACTIVE,
+        );
+      } else {
+        key.firstElementChild.classList.remove(
+          isPlatformWindows() ? classes.META_WIN_ACTIVE : classes.META_APPLE_ACTIVE,
+        );
+        key.firstElementChild.classList.add(
+          isPlatformWindows() ? classes.META_WIN : classes.META_APPLE,
+        );
+      }
+      break;
+    } else if (isSpecialKey(keyText, e)) {
+      addRemoveKeyActive(key);
+      target = key;
+      break;
+    } else if (isKeydown && !e.repeat
+      && (keyText === variables.specialKeys.CAPS_LOCK && e.key === 'CapsLock')) {
```

Comment:
I suggest you to move this condition to meaningful function or variable outside this else if.

Reply:
Fixed like this:
```js
    const isCapsLock = (text, event) => text === variables.specialKeys.CAPS_LOCK
      && event.key === 'CapsLock';
```